pkg/util/cluster/cluster.go

```diff
@@ -168,6 +181,21 @@ func (c *Cluster) Create(ctx context.Context, vnetResourceGroup, clusterName str
 		return err
 	}
 
+	kvName := sharedKeyVaultName
+	if c.ci {
+		// name is limited to 24 characters, but must be globally unique, so we generate one, try if it is availablet
+		kvName := "kv-" + uuid.Must(uuid.NewV4()).String()[:21]
+		result, err := c.keyVaults.CheckNameAvailability(ctx, mgmtkeyvault.VaultCheckNameAvailabilityParameters{Name: &kvName, Type: to.StringPtr("Microsoft.KeyVault/vaults")})
+		if err != nil {
+			return err
+		}
+		if result.NameAvailable != nil && !*result.NameAvailable {
+			return fmt.Errorf("Could not generate unique key vault name: %v", result.Reason)
+		}
+	}
+
+	diskEncryptionSetName := vnetResourceGroup + "-des"
```

Comment:
is this for read DES key?

Reply:
@mjudeikis I don't understand what you mean here. Could you please clarify?

This is the disk encryption set (DES) name which is going to be used for cluster creation with disk encryption.